day_5_supply_stacks/stack_parser.rb

```diff
@@ -0,0 +1,78 @@
+require 'minitest/autorun'
+
+class StackParser
+  def initialize(stack_input)
+    @stack_input = stack_input
+    @stack_rows = stack_rows
+  end
+
+  def stack_columns
+    raw_stack_columns = transpose_stack_rows
+    remove_empty_crates_from(raw_stack_columns)
+  end
+
+  private
+
+  attr_reader :stack_input, :stack_row
+
+  def transpose_stack_rows
```

Comment:
btw. I remembered correctly that you can totally transpose a non-nxn matrix. E.g.:

```ruby
[[1,2,3],[4,5,6]].transpose
=> [[1, 4], [2, 5], [3, 6]]
```

What you can NOT do is:

```ruby
[[1,2,3],[4,5]].transpose
=> `transpose': element size differs (2 should be 3) (IndexError)
```

i.e. the arrays have to be the same size. If you ensure that, you can totally use `Array#transpose` here.

I guess in your case, this was not true - but I'm pretty sure it's easier to ensure that all the arrays are of the same size, than writing your own transpose function (which was a good idea for practicing though!)

Reply:
I was thinking about it (making the arrays the same size), but for some reason, I stuck with this solution.

Making the arrays the same size will also simplify the cleaning up the final matrix (no more `nil` values ).

It seems like a reasonable/worth change.